src/main/scala/mesosphere/marathon/event/http/HttpEventStreamActor.scala

```diff
@@ -16,49 +22,117 @@ trait HttpEventStreamHandle {
+      warnAboutUnknownMessages
+    )
   }
 
-  def addHandler(handle: HttpEventStreamHandle): Unit = {
-    log.info(s"Add EventStream Handle as event listener: $handle. Current nr of listeners: ${clients.size}")
-    val actor = context.actorOf(Props(
-      classOf[HttpEventStreamHandleActor],
-      handle,
-      eventStream,
-      maxOutstandingMessages), handle.id)
-    context.watch(actor)
-    clients += handle -> actor
+  // behaviour components
+
+  /** Immediately close new connections. */
+  private[this] def rejectingNewConnections: Receive = {
+    case HttpEventStreamConnectionOpen(handle) =>
+      log.warn("Ignoring open connection request. Closing handle.")
+      Try(handle.close())
+  }
+
+  /** Accept new connections and create an appropriate handler for them. */
+  private[this] def acceptingNewConnections: Receive = {
+    case HttpEventStreamConnectionOpen(handle) =>
+      metrics.numberOfStreams.setValue(streamHandleActors.size)
+      log.info(s"Add EventStream Handle as event listener: $handle. Current nr of streams: ${streamHandleActors.size}")
+      val actor = context.actorOf(handleStreamProps(handle), handle.id)
+      context.watch(actor)
+      streamHandleActors += handle -> actor
+  }
+
+  /** Switch behavior according to leadership changes. */
+  private[this] def handleLeadership: Receive = {
+    case LeadershipStateEvent.Standby =>
+      log.info("Now standing by. Closing existing handles and rejecting new.")
+      streamHandleActors.values.foreach(context.stop)
+      context.become(standby)
+
+    case LeadershipStateEvent.ElectedAsLeader =>
+      log.info("Became active. Accepting event streaming requests.")
+      context.become(active)
   }
 
-  def removeHandler(handle: HttpEventStreamHandle): Unit = {
-    clients.get(handle).foreach { actor =>
+  /** Cleanup child actors which are not needed anymore. */
+  private[this] def cleanupHandlerActors: Receive = {
+    case HttpEventStreamConnectionClosed(handle) => removeHandler(handle)
+    case Terminated(actor)                       => removeHandlerActor(actor)
+  }
+
+  private[this] def removeHandler(handle: HttpEventStreamHandle): Unit = {
+    streamHandleActors.get(handle).foreach { actor =>
       context.unwatch(actor)
       context.stop(actor)
-      clients -= handle
-      log.info(s"Removed EventStream Handle as event listener: $handle. Current nr of listeners: ${clients.size}")
+      streamHandleActors -= handle
+      metrics.numberOfStreams.setValue(streamHandleActors.size)
+      log.info(s"Removed EventStream Handle as event listener: $handle. " +
+        s"Current nr of listeners: ${streamHandleActors.size}")
     }
   }
 
-  def removeActor(actor: ActorRef): Unit = {
-    clients.find(_._2 == actor).foreach {
+  private[this] def removeHandlerActor(actor: ActorRef): Unit = {
+    streamHandleActors.find(_._2 == actor).foreach {
       case (handle, ref) =>
         log.error(s"Actor terminated unexpectedly: $handle")
```

Comment:
This is no longer unexpected, since it is used when this actor becomes Standby. Either change the logic in the standby case to have the logic here as exception (that was my original intend) or change the log statement.